### ISSUE TITLE
`assign_coords` on xarray datasets breaks `plot_map`

### ISSUE THREAD

**Issue by tbennun** [ecmwf/climetlab#25]
First of all, thanks for developing CliMetLab. It is very convenient and useful!

I'm trying to make a crop that wraps around the longitudinal coordinates, and have been facing a few issues. So far, I've been converting the CML dataset to xarray in order to do so. 
```python
import climetlab as cml
# Dataset probably doesn't matter, but to get the plugin run `pip install climetlab-maelstrom-ens10`
cmlds = cml.load_dataset('maelstrom-ens10', date='20170101', dtype='sfc')
ds = cmlds.to_xarray()
```

Then, I tried to select and re-sort the data using `isel`/`sel`:
```python
cropds = ds.isel(latitude=slice(0,60), longitude=list(range(-80, 0)) + list(range(60)))
```
If I do so, `cml.plot_map(cropds.t2m[0,0,0,0])` yields the following result:
![badcrop-isel](https://user-images.githubusercontent.com/8348955/128509740-db4f64f7-226a-49d1-8693-0a59c4939d46.png)
which is visibly broken in the middle part. Checking with `xarray`'s methods their plot is also broken.

Since the above is probably not the best way, I tried the xarray-recommended `assign_coords`:
```python
dsc = ds.assign_coords(longitude=(((ds.longitude + 180) % 360) - 180))
dsc = dsc.sortby('longitude')
```
In this case, `xarray` itself produces correct plots:
```python
dsc.t2m[0,0,0,0].plot.pcolormesh()
```
![goodcrop-assigncoords](https://user-images.githubusercontent.com/8348955/128510142-6634ac40-3603-48e0-b456-86cb510ac124.png)

but `cml.plot_map` returns an exception:
```
AssertionError                            Traceback (most recent call last)
----> 1 cml.plot_map(dsc.t2m[0,0,0,0])
~/anaconda3/envs/py38/lib/python3.8/site-packages/climetlab/plotting/__init__.py in plot_map(data, **kwargs)
--> 111     p.plot_map(data)
~/anaconda3/envs/py38/lib/python3.8/site-packages/climetlab/plotting/__init__.py in plot_map(self, data, **kwargs)
---> 72             d = get_wrapper(d)
~/anaconda3/envs/py38/lib/python3.8/site-packages/climetlab/wrappers/__init__.py in get_wrapper(data, *args, **kwargs)
---> 52         wrapper = h(data, *args, **kwargs)
~/anaconda3/envs/py38/lib/python3.8/site-packages/climetlab/wrappers/xarray.py in wrapper(data, *args, **kwargs)
--> 110             return XArrayDatasetWrapper(data.to_dataset(), *args, **kwargs)
~/anaconda3/envs/py38/lib/python3.8/site-packages/climetlab/wrappers/xarray.py in __init__(self, data)
     45         if latitude is None or longitude is None:
---> 46             assert latitude is None and longitude is None
```

Thanks in advance!

**Comment by floriankrb** [comment]
Thanks for reporting this issue.

CliMetLab currently relies on metadata to find the lat/lon (through the attribute "standard_name"="longitude"). Using assign_coords removes this metadata and the longitude dimension is not found anymore. 

This issue should be fixed in the next release (if you feel brave, you can test it using the **unstable** develop branch).

**Comment by tbennun** [comment]
Thanks! I can confirm that with the `develop` branch, it now plots properly.

Last question: I didn't know about this metadata, and it may be that other functionality will be lost due to it being omitted. Is there a way to keep it somehow?

**Comment by floriankrb** [comment]
This was a bug in CliMetLab, you shouldn't have to worry about this metadata. CliMetLab should understand this: if coordinate is named "latitude", it is a good candidate to be used as the latitude to plot a map. Any loss of functionality in this respect is likely to be a bug to fix.
(if you really want to keep the metadata you can of course set the attribute manually as ```ds.['latitude'].attrs['standard_name'] = 'latitude'```).

**Comment by floriankrb** [comment]
This should be fixed in the latest release (0.8.18), please feel free to reopen if needed.
